### PR TITLE
Refactor for sub-commands.

### LINKED ISSUE #5: Refactor into subcommands
The current cmdline arg structure is likely to get too unwieldy soon... I'd suggesting refactoring into subcommands, so we'd have something like:

```
# run init/index hook
run prepare --image ... --tag foo_initialized

# run search hook
run search --topics ...
```

Thoughts?

### PR DESCRIPTION
**Pull request by r-clancy**
Fixes #5

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -24,35 +26,36 @@ def build_image():
...
-    parser.add_argument("--topic_format", default="TREC", type=str, help="the topic file for search")
-    parser.add_argument("--qrels", required=True, type=str, help="the qrels file for evaluation")
-    parser.add_argument("--build", default=False, type=bool, help="whether we re-build the image from scratch")
+    # The first step is to pull an image from an OSIRRC participant,
+    # start up a container, run its `init` and `index` hooks, and then
+    # use `docker commit` to save the image after the index has been
+    # built. The rationale for doing this is that indexing may take a
+    # while, but only needs to be done once, so in essence we are
+    # "snapshotting" the system with the indexes.
+    base = client.containers.run("{}:{}".format(args.repo, args.tag), command="sh -c '/init; /index {}'".format(args.collection_name), volumes=volumes, detach=True)
```
Comment (charW) [inline run.py]: Error handling?

Comment (charW) [reply run.py]: We don't have to include error handling in this PR, but it's always good to start thinking about how to resolve errors raised by the Docker Engine SDK.

Comment (r-clancy) [reply run.py]: For sure - this was done quickly in order to get feedback from the mailing list.

File: run.py [hunk 2]
```diff
@@ -24,35 +26,36 @@ def build_image():
...
-    args = parser.parse_args()
 
-    # Create Docker client
-    client = docker.from_env()
+def search():
+    """
+    Runs the search and evaluates the results (run files placed into the /output directory) using trec_eval
+    """
+    exists = len(client.images.list(filters={"reference": "{}:{}".format(args.repo, "save")}))
+    if not exists:
```
Comment (charW) [inline run.py]: For sake of clarify, I would write something like `exists = len(...) != 0`, but it's just personal preference.

**Pull request merged**